docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+  type: "char_buf"
+  returnCopy: true
+  sizeArgIndex: 3
+- index: 2
+  type: "size_t"
+```
+This tells the printer to skip printing the `int` arg because it's not useful.
+
+### Selectors
+
+A `TracingPolicy` can contain from 0 to 5 selectors. A selector is composed of
+1 or more filters. The available filters are the following:
+- [`matchArgs`](#arguments-filter)
+- [`matchReturnArgs`](#return-args-filter)
+- [`matchPIDs`](#pids-filter)
+- [`matchBinaries`](#binaries-filter)
+- [`matchNamespaces`](#namespaces-filter)
+- [`matchCapabilities`](#capabilities-filter)
+- [`matchNamespaceChanges`](#namespace-changes-filter)
+- [`matchCapabilityChanges`](#capability-changes-filter)
+- [`matchActions`](#actions-filter)
+
+#### Arguments filter
+
+Arguments filters can be specified under the `matchArgs` field and provide
+filtering based on the value of the function's argument.
+
+In the next example, a selector is defined with a `matchArgs` filter that tells
+the BPF code to process only the function call for which the second argument,
+index equal to 1, concerns the file under the path `/etc/passwd` or
+`/etc/shadow`. It's using the operator `Equal` to match against the value of
+the argument.
+
+Note that conveniently, we can match against a path directly when the argument
+is of type `file`.
+
+```yaml
+selectors:
+- matchArgs:
+  - index: 1
+  operator: "Equal"
+  values:
+  - "/etc/passwd"
+  - "/etc/shadow"
+```
+
+**Further examples**
+
+In the previous example, we used the operator `Equal`, but we can also use the
+`Prefix` operator and match against all files under `/etc` with:
+```yaml
+selectors:
+- matchArgs:
+  - index: 1
+  operator: "Prefix"
+  values:
+  - "/etc"
+```
+In this situation, an event will be created every time a process tries to
+access a file under `/etc`.
+
+Although it makes less sense, you can also match over the first argument, to
+only detect events that will use the file descriptor 4, which is usually the
+first that come afters stdin, stdout and stderr in process. And combine that
+with the previous example.
+
+```yaml
+- matchArgs:
+  - index: 0
+    operator: "Equal"
+    values:
+    - "3"
+  - index: 1
+    operator: "Prefix"
+    values:
+    - "/etc"
+```
+
+#### Return args filter
+
+Arguments filters can be specified under the `returnMatchArgs` field and
+provide filtering based on the value of the function return value. It allows
+you to filter on the return value, thus success, error or value returned by a
+kernel call.
+
+```yaml
+matchReturnArgs:
+- operator: "NotEqual"
+  values:
+  - 0
+```
+
+#### PIDs filter
+
+PIDs filters can be specified under the `matchPIDs` field and provide filtering
+based on the value of host pid of the process. For example, the following
+`matchPIDs` filter tells the BPF code that observe only hooks for which the
+host PID is equal to either `pid1` or `pid2` or `pid3`:
+
+```yaml
+- matchPIDs:
+  - operator: "In"
+    followforks: true
+    values:
+    - "pid1"
```

Comment:
What is the limit on values?

Reply:
The number of pids you can put? I don't know and I think we should test this.

Reply:
I'd ask in the tetragon channel :)